src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialServiceSupport.scala

```diff
@@ -148,37 +151,64 @@ trait TrialServiceSupport extends LazyLogging {
+\   */
+  def enableSelfForFreeCredits(userInfo: UserInfo): Future[UserTrialStatus] = {
+    thurloeDao.getTrialStatus(userInfo.id, userInfo) flatMap { userTrialStatus =>
+      val doTransition = Enabled.isAllowedFrom(userTrialStatus.state)
+      val newStatus = userTrialStatus.copy(state = Some(Enabled))
+      if (doTransition)
+        thurloeDao.saveTrialStatus(userInfo.id, userInfo, newStatus) map { _ => newStatus }
+      else
+        Future.failed(new FireCloudException(s"User '${userInfo.userEmail} is in state ${userTrialStatus.state} and cannot be enabled."))
+    }
+  }
 
+  // used when campaign manager enables an end user
   def buildEnableUserStatus(userInfo: WorkbenchUserInfo, currentStatus: UserTrialStatus): UserTrialStatus = {
     val needsProject = currentStatus.state match {
       case None | Some(Disabled) => true
       case _ => false // either an invalid transition or noop
     }
 
     if (needsProject) {
-      // how many times should we try to find an available project? Retries here help when multiple users are enabled at once
-      val numAttempts = 50
-
-      def claimProject(attempt:Int):TrialProject = {
-        Try(trialDAO.claimProjectRecord(WorkbenchUserInfo(userInfo.userSubjectId, userInfo.userEmail))) match {
-          case Success(s) => s
-          case Failure(f) =>
-            if (attempt >= numAttempts) {
-              throw new FireCloudException(s"Could not claim a project while enabling user ${userInfo.userSubjectId} " +
-                s"(${userInfo.userEmail}):", f)
-            } else {
-              logger.debug(s"buildEnableUserStatus retrying claim; attempt $attempt")
-              claimProject(attempt + 1)
-            }
-        }
-      }
-      val trialProject = claimProject(1)
-
+      val trialProject = claimProjectWithRetries(userInfo)
       logger.info(s"[trialaudit] assigned user ${userInfo.userEmail} (${userInfo.userSubjectId}) to project ${trialProject.name.value}")
       currentStatus.copy(userId = userInfo.userSubjectId, state = Some(Enabled), userAgreed = false, enabledDate = Instant.now, billingProjectName = Some(trialProject.name.value))
     } else {
       currentStatus.copy(state = Some(Enabled))
     }
   }
 
+  /**
+    *
+    * @param userInfo user for whom to claim project
+    * @param numAttempts how many times should we try to find an available project? Retries here help when multiple users are enabled at once
+    * @return claimed project
+    */
+  def claimProjectWithRetries(userInfo: WorkbenchUserInfo, numAttempts: Int = 50): TrialProject = {
+    // log a warning, which should be seen by dev team, if the project pool is running low.
+    val numAvailable:Long = trialDao.countProjects.getOrElse("available", 0L)
+    if (numAvailable < FireCloudConfig.Trial.projectBufferSize)
+      logger.error(s"There are only $numAvailable free trial projects available; create more as soon as possible!")
```

Comment:
Can we just... do this programmatically from here? It won't help the user that's currently trying to claim because the request will time out before the project creates, but it would help subsequent users

Reply:
Alternatively, we could spin off an async request that requests `n` projects so the pool never gets too low - that would not interfere with the current user's claim process. However, I lean towards keeping this "manual-intervention-required" for the sake of keeping tight control over trial project creation.

Reply:
the problem with triggering creation of new projects here is that:
* project creation via `ProjectManager` is intentionally throttled to avoid hitting quotas, especially because end users may be creating projects at the same time
* we have two instances of orch in production
* there's currently no coordination between orch instances on project creation

So, if a user hits this code on instance 1 and triggers creation of N projects, then a second user hits instance 2 and triggers another creation of N projects, we have both instances spinning and we're creating at 2x our throttle. I'd love to fix this for other reasons but I think it's out of scope of this PR.

Reply:
Agree that it is out of scope!